src/create_disk.sh

```diff
@@ -30,14 +31,83 @@ Options:
     --help: show this help
     --kargs: kernel CLI args
     --no-x86-bios-bootloader: don't install BIOS bootloader on x86_64
+    --with-secure-execution:  generate qcow2 image with IBM SecureExecution feature enabled
 
 You probably don't want to run this script by hand. This script is
 run as part of 'coreos-assembler build'.
 EOC
 }
 
+#SecureExecution variables
+secure_execution_hostkey="/dev/disk/by-id/virtio-hostkey"
+secure_execution_initrd="/tmp/initrd.secure"
+secure_execution_bootfs_key="boot"
+secure_execution_bootfs_label="crypt_bootfs"
+secure_execution_bootfs_device="/dev/mapper/crypt_bootfs"
+
+secure_execution_rootfs_key="root"
+secure_execution_rootfs_label="crypt_rootfs"
+secure_execution_rootfs_device="/dev/mapper/crypt_rootfs"
+
+regenerate_initramfs() {
+    old_initrd=$1
+    new_initrd=$2
+    # Unpacking existing initramdisk
+    currdir=$PWD
+    workdir=$(mktemp -d -p /tmp se-initramfs-XXXXXX)
+    cd ${workdir}
+    gzip -cd ${old_initrd} | cpio -imd --quiet
+    # Adding LUKS root key and crypttab config
+    mkdir -p etc/luks
+    cp -f $deploy_root/etc/luks/* etc/luks/
+    cp -f $deploy_root/etc/crypttab etc/
+    # Creating new initramdisk image
+    find . -mindepth 1 | cpio --quiet -H newc -o | gzip -9 -n >> ${new_initrd}
+    # Cleanup
+    cd ${currdir}
+    rm -rf ${workdir}
+}
+
+propogate_luks_config() {
+    key="$1"
+    lbl="$2"
+    # Moving key file to final destination
+    mkdir -p $deploy_root/etc/luks
+    mv /tmp/$key $deploy_root/etc/luks/$key
+    chmod 0400 $deploy_root/etc/luks/$key
+    #
+    uuid=$(cryptsetup luksUUID /dev/disk/by-label/$lbl)
+    #
+    if [[ ! -e $deploy_root/etc/crypttab ]]; then
+        touch $deploy_root/etc/crypttab
+        chmod 0600 $deploy_root/etc/crypttab
+    fi
+    echo "$lbl UUID=${uuid} /etc/luks/$key luks" >> $deploy_root/etc/crypttab
+}
+
+create_luks_partition() {
+    key="/tmp/$1"
+    lbl="$2"
+    dev="$3"
+    # Generationg random key
+    dd if=/dev/urandom of=$key bs=1024 count=4
+    chmod 0400 $key
+    cryptsetup luksFormat -q \
+                        --type luks2 \
+                        --label="$lbl" \
+                        --key-file=$key \
```

Comment:
SE uses LUKS for integrity as well as confidentiality, so I think we need `--integrity` here?

Reply:
right now with `--integrity` system fails during `firstboot`, so i skipped that in Step2

Reply:
Why does it fail?